modules/@apostrophecms/admin-bar/ui/apos/components/TheAposContextBar.vue

```diff
@@ -40,17 +40,19 @@ export default {
   name: 'TheAposContextBar',
   mixins: [ AposPublishMixin, AposAdvisoryLockMixin ],
   data() {
-    // Since cookie-based login sessions and sessionStorage are not precisely the same
-    // thing, correct any forbidden combination at page load time
-    if (window.apos.mode === 'published') {
-      window.sessionStorage.setItem('aposEditMode', JSON.stringify(false));
+    const query = apos.http.parseQuery(location.search);
+    // If the URL references a draft, go into draft mode but then clean up the URL
+    const draftMode = query['apos-mode'] || 'published';
+    if (draftMode === 'draft') {
+      delete query['apos-mode'];
+      history.replaceState(null, '', apos.http.addQueryToUrl(location.href, query));
```

Comment:
This query parameter got here via the second chance redirect in the page module, we needed it to load the URL with draft content, but now we can remove it from the address bar so folks don't get stuck sharing / linking to URLs with an inappropriate query parameter for most audiences.